src/lib/routers/history.ts

```diff
@@ -129,7 +131,13 @@ class BrowserHistory implements Router {
 
     this.writeTimer = window.setTimeout(() => {
       setWindowTitle(title);
-      window.history.pushState(routeState, title || '', url);
+
+      if (replace) {
+        window.history.replaceState(routeState, title || '', url);
```

Comment:
this means the last search state will be removed in regular use cases where you dispose the InstantSearch:

```
mount: ?
query: ?query=123 --> deleted at dispose
dispose: ?
```

Reply:
Actually in SPA implementation, this breaks in the same way. What should be an option is *what to do* for writing after dispose happens,

1. vanilla, where dispose is on the same page: write on dispose
2. angular, vue etc. with Vue Router, where dispose happens together with a page change: no write after disposed

Reply:
> 1. vanilla, where dispose is on the same page: write on dispose

do you mean write with `replaceState`? I don't see many uses cases for dispose in vanilla not pushing a new state on dispose as it was a navigation event we may want to go back to sounds fair to me.

> 2. angular, vue etc. with Vue Router, where dispose happens together with a page change: no write after disposed

So that would mean exposing an option removing the final `write({ })` altogether?
Sounds fair to me. 

```js
history({ preventWriteOnDispose: true })
```


```js
dispose() {
    ...
    if (!preventWriteOnDispose) {
        write({ })
    }
   _diposed = true;
},

write(routeState) {
   if (_diposed) return // we shouldn't be able to write after dispose
   ...
}
```

We'd still need to make sure we're no longer able to make writes after `dipose` call.
We can either do that here or on the middleware level (ensuring `onStateChange` has no effect after `unsubscribe` has been called). This sounds like a more legit change.